From 1d65ac128dde0988d6e69ed0f2ffaa8673bf3ce3 Mon Sep 17 00:00:00 2001
From: John Barberiz <jbarberi@amd.com>
Date: Mon, 15 Jan 2018 18:30:22 -0500
Subject: [PATCH 4762/5855] drm/amd/display: Add temporary delay on disconnect
 patch

- Sequencing bug of not resetting delay on disconnect to default values

Signed-off-by: John Barberiz <jbarberi@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc_link.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc_link.c b/drivers/gpu/drm/amd/display/dc/core/dc_link.c
index c7e159a..420bc2d 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc_link.c
@@ -700,6 +700,8 @@ bool dc_link_detect(struct dc_link *link, enum dc_detect_reason reason)
 		if (sink->edid_caps.panel_patch.disconnect_delay > 0 &&
 				SIGNAL_TYPE_HDMI_TYPE_A)
 			program_hpd_filter(link, sink->edid_caps.panel_patch.disconnect_delay);
+		else
+			program_hpd_filter(link, DEFAULT_DELAY_DISCONNECT);
 
 		/* HDMI-DVI Dongle */
 		if (sink->sink_signal == SIGNAL_TYPE_HDMI_TYPE_A &&
-- 
2.7.4

